app-typescript/components/ToastMessage.tsx

```diff
@@ -0,0 +1,94 @@
+import * as React from 'react';
+import ToastText from './ToastText';
+import classNames from 'classnames';
+
+export type MessageProp =
+    | React.ReactNode
+    | string;
+
+export type Position = 'top' | 'bottom' | 'top-right' | 'top-left' | 'bottom-right' | 'bottom-left';
+
+export type NotesType = 'default' | 'primary' | 'success' | 'warning' | 'alert' | 'highlight' | 'light';
+
+export interface IMessageOptions {
+    id?: string;
+    duration?: number | null;
+    type?: NotesType;
+    position?: Position;
+    icon?: string;
+    show?: boolean;
+}
+
+interface IProps extends IMessageOptions {
+    message: MessageProp;
+    closeElement(id, position): void;
+}
+
+interface IState {
+    show: boolean;
+}
+
+ export const ToastMessage = ({
```

Comment:
just wondering why it has the message property instead of just using children?
```ts
<ToastMessage>{'some text here'}</ToasMessage>
```
that would be more intuitive imo which is important for a library

Reply:
This is an internal component. Here's the exported function which would be called by consumers of the library: https://github.com/superdesk/superdesk-ui-framework/pull/436/files#diff-82aa67b92768a0ff22d1519a15c0e432R32